### ISSUE TITLE
Add new 'AI or climate?' question on submissions

### ISSUE THREAD

**Issue by hellafitz** [Iridescent-CM/technovation-app#2831]
Add a new question to the submissions about whether the team's submissions uses AI or addresses climate change. The goal is for our staff to be able to search and pull out qualifying submissions for new prizes. Work with Rebecca on exact language and placement.

PT will be communicating with students
Timing: sometime in next couple of iterations (submissions won’t really come in for a bit)
We want to be able to search submissions to find eligible teams

**Comment by hellafitz** [comment]
@rsgonzal @ksrebecca This would be a good one for you to collab on during Friday's TF/TG. Here's an Invision board for mocking up the changes. https://projects.invisionapp.com/freehand/document/zZCtgCu0O

**Comment by rsgonzal** [comment]
Invisionapp updated with mockups

**Comment by hellafitz** [comment]
@ksrebecca should the AI / climate Qs section be included in the % completed metric for in-progress submissions? If so, how much and where should we lower the other section %s?

**Comment by ksrebecca** [comment]
no because it's optional, not everyone will check a box...

**Comment by rsgonzal** [comment]
Thanks, @ksrebecca ! So the Team Photo is also one that isn't required and doesn't affect the progress. Maybe this one can work like that?

**Comment by shaunxp20** [comment]
Do we want these questions/answers to show up on the submission summary page (which appears right before submitting, and when clicking the "View your complete submission" button)?

**Comment by rsgonzal** [comment]
> Do we want these questions/answers to show up on the submission summary page (which appears right before submitting, and when clicking the "View your complete submission" button)?

Great questions, Shaun! I say yes. We don't want to confuse the students since they've entered this information.

**Comment by ksrebecca** [comment]
Agree with Rox on that :D

**Comment by shaunxp20** [comment]
Okay, this one is on the QA site for a first look! 😅 I added some QA steps for things to look for too. 😃

**Comment by ksrebecca** [comment]
QA'ing:

View the new "Additional Information" section (My Team's Submission -> Ideation)
Verify the page/content looks good, it works as expected with the dropdowns

- Yes!

Verify the answers get saved, and look good on the Ideation summary page

- Everything got saved EXCEPT the description answer i typed in for the game. I tried a longer and a shorter description but it never showed up on the ideation summary page
<img width="504" alt="Screen Shot 2021-03-09 at 7 22 10 PM" src="https://user-images.githubusercontent.com/43972336/110556639-2c671880-80fc-11eb-937a-c9899e4f10be.png">
<img width="485" alt="Screen Shot 2021-03-09 at 7 22 34 PM" src="https://user-images.githubusercontent.com/43972336/110556645-2ffa9f80-80fc-11eb-9441-91cd99735f04.png">

Verify the questions/answers look good on the submission summary page

- All look good except the elaboration on the game question that doesn't appear there either.
<img width="901" alt="Screen Shot 2021-03-09 at 7 25 08 PM" src="https://user-images.githubusercontent.com/43972336/110556815-836ced80-80fc-11eb-8200-a34432762c66.png">

Verify a submission can be completed without filling out these new questions (they're not required)

- Yes. When I changed all the answers back to original null fields it still said the app was submitted into judging even though I didn't have answers for those questions :)
<img width="791" alt="Screen Shot 2021-03-09 at 7 27 49 PM" src="https://user-images.githubusercontent.com/43972336/110556981-e494c100-80fc-11eb-87c1-aa5eb5951ab3.png">

Super clear, simple, looking GREAT!!

**Comment by shaunxp20** [comment]
> Everything got saved EXCEPT the description answer i typed in for the game. I tried a longer and a shorter description but it never showed up on the ideation summary page

> All look good except the elaboration on the game question that doesn't appear there either.

Good catch! I fixed this, it's been deployed to QA so should be good now! 😄

**Comment by rsgonzal** [comment]
quick q @shaunxp20 ! Would your update clear previous answers? just cause it looks like i'm on the same one rebecca updated and it showed like it hadn't been responded to, but then when i went and answered rebeccas reponses were there!

Just making sure this isn't a bug and the responses are saved

**Comment by shaunxp20** [comment]
> quick q @shaunxp20 ! Would your update clear previous answers? just cause it looks like i'm on the same one rebecca updated and it showed like it hadn't been responded to, but then when i went and answered rebeccas reponses were there!
> 
> Just making sure this isn't a bug and the responses are saved

The update wouldn't have cleared previous answers. Before this fix, the answers were being saved, but weren't being displayed - so if there were answers entered/saved before this fix, after this fix they would be displayed. Hopefully that makes sense! 😊 😃

**Comment by rsgonzal** [comment]
> Dropdown worked as expected but I was able to save my answer without adding text explaining how it uses AI or climate... Could we make it required for them to fill out the explanation portion if they select Yes for any of the options?

@shaunxp20 can we make this happen? so if an answer is yes for any of them, they have to fill out the comment box and maybe they get an error message like this if they try to submit without it
![Screen Shot 2021-03-11 at 5 44 52 PM](https://user-images.githubusercontent.com/22801026/110879260-81886300-8291-11eb-8db2-b83a1457f7ea.png)

> Suggestion: Maybe having something to separate those answers from the images could be good. All the rest has a sort of titles except for this portion (Pitch, RPEs...)

How does this look @ophelieh @ksrebecca @shaunxp20 

![Screen Shot 2021-03-11 at 5 35 02 PM](https://user-images.githubusercontent.com/22801026/110879466-e5129080-8291-11eb-9dff-41de0e4ba976.png)

**Comment by rsgonzal** [comment]
> The update wouldn't have cleared previous answers. Before this fix, the answers were being saved, but weren't being displayed - so if there were answers entered/saved before this fix, after this fix they would be displayed. Hopefully that makes sense! 😊 😃

Hmm. So if I look at Christmas Town and log in as Santa Maria, My submission doesn'tshow that I completed that section. Which I think ophelie had submitted for this team. Right @ophelieh ?
![Screen Shot 2021-03-11 at 5 51 20 PM](https://user-images.githubusercontent.com/22801026/110879923-acbf8200-8292-11eb-8be8-c6d7e2ef5585.png)

**Comment by ksrebecca** [comment]
> > Suggestion: Maybe having something to separate those answers from the images could be good. All the rest has a sort of titles except for this portion (Pitch, RPEs...)
> 
> How does this look @ophelieh @ksrebecca @shaunxp20
> 
> ![Screen Shot 2021-03-11 at 5 35 02 PM](https://user-images.githubusercontent.com/22801026/110879466-e5129080-8291-11eb-9dff-41de0e4ba976.png)

I like this a lot better, nice to have at the bottom and separated

**Comment by ophelieh** [comment]
Love it! :)

**Comment by ophelieh** [comment]
@rsgonzal I had entered answers then removed them and submitted. Then I played around with it a little more.

**Comment by shaunxp20** [comment]
Awesome! Thanks for the initial feedback, @ophelieh, great suggestions! Thanks for mocking this up @rsgonzal!

Glad everyone is okay w/ the changes! It's on QA for another looksie! (Making the answers required for a "yes" response, and moving the "Additional Information" section). Happy to make more changes too if we need to. 😃

**Comment by ophelieh** [comment]
Loves the new error message if you don't enter an answer! :) 

Love the move for the additional section! 

Will it look like this for judges as well? Should we test that too?

**Comment by ksrebecca** [comment]
Yes! Error message looks great, works for all Yes answers. The move to the bottom of the submission is great too.

THANK YOU!!

**Comment by rsgonzal** [comment]
Awesome! Agree this looks good!

@ophelieh no need to test for judges! They won't be able to see the additional info!

**Comment by ophelieh** [comment]
oh right! haha Forgot about that. 

Will we be able to pull that information on the submission tab when we need to figure out which ones to select? 
Thank you!! :)

**Comment by shaunxp20** [comment]
Finally got this one deployed, thanks y'all! 🎉 🎊
